compiler-core/src/type_.rs

```diff
@@ -1569,6 +1570,7 @@ fn assert_suitable_main_function(
 ) -> Result<(), crate::Error> {
     let not_found = || crate::Error::ModuleDoesNotHaveMainFunction {
         module: module_name.clone(),
+        origin: Origin::Src,
```

Comment:
This is hard-coded to src, but it could be one of the others origins. It'll need to be passed in.

Reply:
uuupppps 🙃 added as parameter to the assertion, also adapted existing tests using it